### ISSUE TITLE
Refresh values on a specified interval instead of continuous update

### ISSUE THREAD

**Issue by TimofeyK** [hannseman/homebridge-mi-hygrothermograph#35]
Mi sensor reads temperature / humidity values every second and often readings would change for 0.1 every second e.g. 25.2 <-> 25.1

Current behaviour:
The plugin seems to be scanning Bluetooth continuously and reports value changes back to HomeKit every second or so

This causes automation rules / triggers created for temperature change to fire every second

Additional side effect:
- Bluetooth is continuously locked on the host and cannot be  used for any other application
- Potentially shortened battery life of the sensor

Expected behaviour:
Room temperatures do not require monitoring to a tenth of a degree every second so a reasonable or configurable refresh interval for scanning should be implemented

Raspberry Pi Zero W
Raspbian Stretch
Homebridge 0.4.45
NodeJS 9.90
Noble - 1.9.1

**Comment by TimofeyK** [comment]
I certainly can be wrong about “side effects”. I thought Bluetooth is getting locked while “listening” to sensor’s advertisements because I could not use it to scan or connect from another process while Homebridge was running

However I am sure about automations. HomeKit DOES support triggers based on sensor values but stock Home app does not support configuring them. 3rd party HomeKit apps such Eve or Home3 can configure such triggers. I did that just today configuring a http accessory plugin to call my aircon. I was watching how that happened every second or two in the log

https://hochgatterer.me/home/automation/
Here is the reference I followed, see the section “Accessory state trigger”. The trick is to fire the trigger on any temperature change and then evaluate the temperature value and other conditions such aircon state. This definitely works but I find this validation every second to be excessive. Notifying state change to HomeKit with some configured intervals would be a good solution. E.g. 0 for real-time or defined interval in seconds or minutes

Otherwise your plugin is fantastic! Worked from the first attempt and very stable. Thanks for that!

**Comment by hannseman** [comment]
I've implemented this feature in https://github.com/hannseman/homebridge-mi-hygrothermograph/pull/38. Will try and merge this ASAP-

**Comment by andreyxc** [comment]
Thank you for the project. It's not working updateInterval ,updates every second.

**Comment by TimofeyK** [comment]
Yes, it does not seem to work as expected. I installed the new version and set the interval to 60. Then I created a trigger for temperature change with the action that I can easily monitor in the log. It was triggered at random intervals from 2 seconds to something like 40 seconds

Another idea for this function: maybe you can implement more advanced logic and refresh HK value at specified interval or if the value received from the sensor has changed more than the specified threshold e.g. 0.3°. This way you can set longer refresh intervals but still have near real-time monitoring if the temperature changes rapidly, at the same time it will eliminate noise from constant 0.1° changes

**Comment by hannseman** [comment]
@TimofeyK and @andreyxc did you properly install the #38 PR? Because installing from `npm` or the current master will not have the updateInterval functionality.

**Comment by hannseman** [comment]
Just did a bit more testing and it does indeed appear to be working as expected. This is the output from `mqtt_sub` with `updateInterval` set to `10`.

```
2019-02-15 14:31:06.622 home/entrance/sensors/temperature 22.9
2019-02-15 14:31:06.640 home/entrance/sensors/humidity 30.8
2019-02-15 14:31:06.658 home/entrance/sensors/battery 0
2019-02-15 14:31:16.647 home/entrance/sensors/temperature 22.9
2019-02-15 14:31:16.664 home/entrance/sensors/humidity 30.8
2019-02-15 14:31:16.679 home/entrance/sensors/battery 0
2019-02-15 14:31:26.659 home/entrance/sensors/temperature 22.9
2019-02-15 14:31:26.676 home/entrance/sensors/humidity 30.9
2019-02-15 14:31:26.693 home/entrance/sensors/battery 0
2019-02-15 14:31:36.677 home/entrance/sensors/temperature 22.9
2019-02-15 14:31:36.694 home/entrance/sensors/humidity 30.9
2019-02-15 14:31:36.710 home/entrance/sensors/battery 0
```

**Comment by andreyxc** [comment]
Don't understand but the files themselves do not mention updateInterval

**Comment by TimofeyK** [comment]
Ah, sorry. I thought it was already merged so simply installed v2.0 from NPM

**Comment by hannseman** [comment]
> Don't understand but the files themselves do not mention updateInterval
the files are not updateInterval
sorry for the English translation

This is because it has not yet been merged.

> Ah, sorry. I thought it was already merged so simply installed v2.0 from NPM
No problem, I'll merge #38 ASAP and make a 2.1 release.

**Comment by hannseman** [comment]
I've just released 2.1.0. Please give it a try.

**Comment by andreyxc** [comment]
Yes thank you all is well
